src/arrayMethodSort.js

```diff
@@ -4,9 +4,39 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function(compareFunction = compareAsString) {
+    let count, prevValue;
+
+    do {
+      count = 0;
+
+      for (let i = 1; i < this.length; i++) {
+        if (compareFunction(this[i - 1], this[i]) > 0) {
+          count++;
+
+          prevValue = this[i - 1];
+          this[i - 1] = this[i];
+          this[i] = prevValue;
+        }
+      }
+    } while (count > 0);
+
+    return this;
   };
 }
 
+function compareAsString(a, b) {
+  const strA = String(a);
+  const strB = String(b);
+
+  switch (true) {
+    case (strA > strB):
+      return 1;
+    case (strA === strB):
+      return 0;
+    default:
+      return -1;
+  }
```

Comment:
I'm sorry I confused you with my previous comment, but nevertheless, it's better to use `if` statements here. Just no need to use `else`, as your code will not go further after `return` anyway. What about switch...case - it's a great tool when you are switching some variable, but it's better to avoid `switch(true)` as it works the same as if...else  but can be more difficult for other developers to understand.
```suggestion
   if (strA > strB) {
      return 1;
   }
   
   if (strA < strB) {
     return -1;
   }
   
   return 0;
```